package/MDAnalysis/analysis/base.py

```diff
@@ -121,103 +121,21 @@
 tools if only the single-frame analysis function needs to be written.
 
 """
-from collections import UserDict
 import inspect
-import logging
 import itertools
+import logging
+import warnings
+from functools import partial
+from typing import Callable, Iterable, Sequence, Union
 
 import numpy as np
 from MDAnalysis import coordinates
 from MDAnalysis.core.groups import AtomGroup
 from MDAnalysis.lib.log import ProgressBar
 
-logger = logging.getLogger(__name__)
-
-
-class Results(UserDict):
```

Comment:
I assume you moved this class to `parallel.py` to avoid circular dependencies? In hindsight, I think that's not the right place, since one would expect only things related to parallel analysis to be in there. What about creating an `utils.py` or `results.py` module?

Reply:
>I assume you moved this class to parallel.py to avoid circular dependencies
Yes, indeed.

I am fine with creating `results.py`, although it's not that big of a file -- although definitely not `utils.py`, since it's too generic.

Reply:
Resolving, this has been moved.